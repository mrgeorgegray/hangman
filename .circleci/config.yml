version: 2
jobs:
  build:
    filters:
      branches:
        ignore:
          - gh-pages
    docker:
      - image: circleci/node:11.4.0
    steps:
      - checkout
      - restore_cache:
          name: Restore Yarn Package Cache
          keys:
          - yarn-packages-{{ checksum "yarn.lock" }}
      - run: 
          name: Install Dependencies
          command: yarn install --frozen-lockfile --cache-folder ~/.cache/yarn
      - save_cache:
          name: Save Yarn Package Cache
          key: yarn-packages-{{ checksum "yarn.lock" }}
          paths:
            - ~/.cache/yarn
      - run:
          name: "Tests"
          command: yarn test --coverage