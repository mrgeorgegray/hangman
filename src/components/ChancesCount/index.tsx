/** @jsxImportSource @emotion/react */
import { css, useTheme } from "@emotion/react";

interface ChancesCountProps {
  chancesRemaining: number;
}

const ChancesCount: React.FC<ChancesCountProps> = ({ chancesRemaining }) => {
  const { colors, fontSize, space } = useTheme();

  return (
    <div
      css={css`
        font-size: ${fontSize[0]}px;
        margin-bottom: ${space[2]}px;
      `}
    >
      <p
        id="chancesRemaining"
        aria-live="polite"
        css={css`
          color: ${colors.grey};
        `}
      >
        (chances remaining: {chancesRemaining})
      </p>
    </div>
  );
};

export default ChancesCount;
