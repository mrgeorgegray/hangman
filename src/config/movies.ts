/* ===============
MOVIES
all numbers and special chars removed
=============== */
const movies = [
  "a beautiful mind",
  "a charlie brown christmas",
  "a christmas story",
  "a quiet place",
  "a star is born",
  "about time",
  "ad astra",
  "alien",
  "always be my maybe",
  "american beauty",
  "anchorman",
  "annihilation",
  "apocalypto",
  "arrival",
  "arthur christmas",
  "avatar",
  "baby driver",
  "back to the future",
  "bad santa",
  "baywatch",
  "beautiful boy",
  "big hero six",
  "bird box",
  "black panther",
  "black swan",
  "black widow",
  "blade runner",
  "blockers",
  "booksmart",
  "boyhood",
  "bridesmaids",
  "bridget jones’s diary",
  "call me by your name",
  "cast away",
  "chicken run",
  "children of men",
  "chinatown",
  "christmas vacation",
  "clueless",
  "collateral",
  "crazy rich asians",
  "crazy, stupid, love",
  "dawn of the dead",
  "despicable me",
  "die hard",
  "die hard",
  "dodgeball",
  "dolittle",
  "dora and the lost city of gold",
  "dumb and dumber",
  "dumbo",
  "edge of tomorrow",
  "edward scissorhands",
  "elf",
  "eyes wide shut",
  "finding dory",
  "forgetting sarah marshall",
  "forrest gump",
  "free guy",
  "frozen",
  "game night",
  "get out",
  "ghost dog",
  "ghostbusters",
  "gifted",
  "gladiator",
  "good boys",
  "goodfellas",
  "gravity",
  "green book",
  "halloween",
  "hancock",
  "happiest season",
  "heat",
  "hereditary",
  "home again",
  "home alone",
  "how the grinch stole christmas!",
  "how to train your dragon",
  "hustlers",
  "i feel pretty",
  "inception",
  "raiders of the lost ark",
  "inside out",
  "insomnia",
  "interstellar",
  "it",
  "it happened on fifth avenue",
  "james bond: casino royale",
  "jaws",
  "john wick",
  "jojo rabbit",
  "joyeux noël",
  "kill bill",
  "kingsman: the secret service",
  "klaus",
  "knives out",
  "knocked up",
  "kong skull island",
  "kung fu panda",
  "last christmas",
  "lethal weapon",
  "liar liar",
  "lion",
  "logan",
  "love actually",
  "love actually",
  "love, simon",
  "lucky number slevin",
  "mad max",
  "mary poppins returns",
  "meet the fockers",
  "minions",
  "minority report",
  "minority report",
  "mission impossible",
  "moana",
  "monsters inc.",
  "moon",
  "moonlight",
  "mother!",
  "mulholland dr.",
  "munich",
  "my spy",
  "mystic river",
  "no country for old men",
  "noelle",
  "notting hill",
  "old school",
  "oldboy",
  "paranormal activity",
  "parasite",
  "predator",
  "pretty woman",
  "prisoners",
  "rango",
  "ready player one",
  "ring",
  "rocky",
  "rudolph, the red-nosed reindeer",
  "run lola run",
  "saving private ryan",
  "scoob!",
  "scream",
  "scrooge",
  "scrooged",
  "seven",
  "shazam!",
  "shrek",
  "shutter island",
  "silent night",
  "sin city",
  "snowpiercer",
  "sonic the hedgehog",
  "sorry to bother you",
  "soul",
  "source code",
  "southpaw",
  "spider-man",
  "split",
  "spotlight",
  "star trek the wrath of khan",
  "star wars the rise of skywalker",
  "step brothers",
  "straw dogs",
  "suicide squad",
  "superbad",
  "tag",
  "taken",
  "taxi driver",
  "tenet",
  "the accountant",
  "the big lebowski",
  "the big sick",
  "the blair witch project",
  "the bourne identity",
  "the cabin in the woods",
  "the christmas chronicles",
  "the conjuring",
  "the dark knight",
  "the departed",
  "the exorcist",
  "the favourite",
  "the fly",
  "the fugitive",
  "the game",
  "the godfather",
  "the green mile",
  "the half of it",
  "the house with a clock in its walls",
  "the human centipede",
  "the incredibles",
  "the invisible man",
  "the lego movie",
  "the lion king",
  "the longshot",
  "the martian",
  "the matrix",
  "the next three days",
  "the nice guys",
  "the night before",
  "the nightmare before christmas",
  "the notebook",
  "the other guys",
  "the others",
  "the peanut butter falcon",
  "the prestige",
  "the proposal",
  "the santa clause",
  "the secret life of pets",
  "the shawshank redemption",
  "the shining",
  "the shop around the corner",
  "the silence of the lambs",
  "the silence of the lambs",
  "the sixth sense",
  "the terminator",
  "the texas chain saw massacre",
  "the thing",
  "the transporter",
  "the usual suspects",
  "the wailing",
  "the witch",
  "there will be blood",
  "there’s something about mary",
  "thor",
  "titanic",
  "togo",
  "tomb raider",
  "total recall",
  "toy story",
  "trading places",
  "trainwreck",
  "trolls",
  "tropic thunder",
  "twelve monkeys",
  "up",
  "us",
  "v for vendetta",
  "vice",
  "wall-e",
  "wedding crashers",
  "when harry met sally",
  "whiplash",
  "wonder",
  "wreck-it ralph",
  "you’ve got mail",
  "zoolander",
];

export default movies;
