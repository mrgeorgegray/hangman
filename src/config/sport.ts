/* ===============
SPORT
all numbers and special chars removed
=============== */
const sport = [
  "aerobics",
  "archer",
  "archery",
  "arrow",
  "athlete",
  "athletics",
  "badminton",
  "ball",
  "base",
  "baseball",
  "basketball",
  "bat",
  "baton",
  "batter",
  "batting",
  "biathlon",
  "bicycle",
  "bicycling",
  "bike",
  "biking",
  "billiards",
  "bobsleigh",
  "bocce",
  "boomerang",
  "boules",
  "bow",
  "bowler",
  "bowling",
  "boxer",
  "boxing",
  "bronze medal",
  "bunt",
  "canoe",
  "canoeing",
  "catch",
  "catcher",
  "champion",
  "championship",
  "cleats",
  "club",
  "coach",
  "compete",
  "competing",
  "competition",
  "competitor",
  "crew",
  "cricket",
  "croquet",
  "cross country",
  "cue",
  "curling",
  "cycle",
  "cycling",
  "cyclist",
  "dart",
  "dartboard",
  "deadlifting",
  "decathlon",
  "defense",
  "diamond",
  "discus",
  "dive",
  "diver",
  "diving",
  "dodgeball",
  "doubleheader",
  "dugout",
  "epee",
  "equestrian",
  "equipment",
  "exercise",
  "fencing",
  "field",
  "fielder",
  "field hockey",
  "fielding",
  "figure skating",
  "fitness",
  "football",
  "forward",
  "free throw",
  "frisbee",
  "game",
  "gear",
  "geocaching",
  "goal",
  "goalie",
  "golf",
  "golfer",
  "golfing",
  "guard",
  "gym",
  "gymnast",
  "gymnastics",
  "gymnasium",
  "halftime",
  "hammer throw",
  "handball",
  "hang gliding",
  "hardball",
  "helmet",
  "heptathlon",
  "high jump",
  "hitter",
  "hockey",
  "home",
  "home plate",
  "home run",
  "home team",
  "hoop",
  "horseshoes",
  "huddle",
  "hurdle",
  "ice hockey",
  "ice rink",
  "ice skates",
  "ice skating",
  "infield",
  "infielder",
  "inline skates",
  "inning",
  "javelin",
  "jog",
  "jogger",
  "judo",
  "jump",
  "jumper",
  "jumping",
  "jump rope",
  "karate",
  "kayak",
  "kayaker",
  "kayaking",
  "kickball",
  "kneepads",
  "king fu",
  "kite",
  "lacrosse",
  "lawn bowling",
  "league",
  "long jump",
  "lose",
  "loser",
  "luge",
  "major league",
  "mallet",
  "martial arts",
  "mat",
  "medal",
  "minor league",
  "mitt",
  "mouthguard",
  "move",
  "movement",
  "net",
  "nordic skiing",
  "offense",
  "ollie",
  "orienteering",
  "out",
  "outfield",
  "outfielder",
  "paddle",
  "paddleball",
  "paddling",
  "paintball",
  "parasailing",
  "parkour",
  "pentathlon",
  "pickleball",
  "ping pong",
  "pitch",
  "pitcher",
  "play",
  "player",
  "playing",
  "playoffs",
  "pogo stick",
  "pole",
  "pole vault",
  "polo",
  "pool",
  "puck",
  "quarter",
  "quarterback",
  "quiver",
  "race",
  "racer",
  "racewalking",
  "racing",
  "racket",
  "racquetball",
  "rafting",
  "referee",
  "relay",
  "ride",
  "riding",
  "rink",
  "rock climbing",
  "roller skates",
  "roller skating",
  "row",
  "rower",
  "rowing",
  "rugby",
  "run",
  "runner",
  "running",
  "sailing",
  "score",
  "scoreboard",
  "scuba",
  "scull",
  "sculling",
  "shortstop",
  "shot put",
  "silver medal",
  "skate",
  "skating rink",
  "skeleton",
  "ski",
  "skier",
  "skiing",
  "slalom",
  "sled",
  "sledder",
  "sledding",
  "snorkling",
  "snowboard",
  "snowboarder",
  "snowboarding",
  "snowshoeing",
  "soccer",
  "softball",
  "somersault",
  "speed skating",
  "sport",
  "sportsmanship",
  "squash",
  "stadium",
  "stick",
  "strike",
  "stroke",
  "sumo wrestling",
  "surfer",
  "surfing",
  "swim",
  "swimmer",
  "swimming",
  "synchronized swimming",
  "table tennis",
  "taekwondo",
  "tag",
  "target",
  "team",
  "teammate",
  "tee",
  "telemark skiing",
  "tennis",
  "tetherball",
  "throw",
  "throwing",
  "tie",
  "toboggan",
  "track and field",
  "trampoline",
  "triathlon",
  "triathlete",
  "triple jump",
  "triple play",
  "tug of war",
  "ultramarathon",
  "umpire",
  "unicycle",
  "unicyclist",
  "uniform",
  "vaulter",
  "vaulting",
  "visiting team",
  "volley",
  "volley ball",
  "wakeboarding",
  "walk",
  "walker",
  "walking",
  "water polo",
  "waterski",
  "waterskier",
  "waterskiing",
  "weightlifter",
  "weightlifting",
  "weights",
  "wetsuit",
  "white water rafting",
  "wicket",
  "win",
  "windsurfer",
  "windsurfing",
  "winner",
  "winning",
  "wrestler",
  "wrestling",
];

export default sport;
