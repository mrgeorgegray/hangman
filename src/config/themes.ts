import { Theme } from "@emotion/react";

const baseTheme: Theme = {
  border: {
    color: "#30353B",
    radius: 3,
    width: 2,
  },
  breakpoints: {
    md: "500px",
    lg: "720px",
  },
  colors: {
    background: "#F5F5F5",
    error: "#FF0033",
    grey: "#949DA5",
    link: "#0366D6",
    linkHover: "#0387D6",
    purple: "#6E40C9",
    softError: "#FFBABA",
    softSuccess: "#9BE9A8",
    success: "#27D545",
    text: "#30353B",
    warning: "#FF7903",
  },
  fontSize: [14, 18, 22, 34, 50],
  hangman: {
    fill: "#30353B",
    stroke: "48, 53, 59",
  },
  layout: {
    maxWidth: "1000px",
  },
  space: [4, 8, 16, 32, 64, 96, 128],
  transitions: [0.15, 0.2, 0.5],
};

export const lightTheme: Theme = {
  ...baseTheme,
};

export const darkTheme: Theme = {
  ...baseTheme,
  border: {
    ...baseTheme.border,
    color: "transparent",
  },
  colors: {
    ...baseTheme.colors,
    background: "#0D1117",
    link: "#58A6FF",
    linkHover: "#ABE0FF",
    softError: "#DA3733",
    softSuccess: "#02602C",
    text: "#F5F5F5",
  },
  hangman: {
    fill: "#ADADAD",
    stroke: "245, 245, 245",
  },
};
