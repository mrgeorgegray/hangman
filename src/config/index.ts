import movies from "./movies";
import sport from "./sport";
import words from "./words";
import { darkTheme, lightTheme } from "./themes";

/* ===============
STARTING_CHANCES
how many guesses a user
has before the Hangman is dead
must match <Hangman /> setup.
=============== */
export const STARTING_CHANCES = 6;

/* ===============
LETTERS
determine what is displayed
on the keyboard. Make sure
you include all letters that
are in your solutions.
=============== */
// prettier-ignore
export const LETTERS = [
  "a", "b", "c", "d", "e", "f", "g", "h", "i",
  "j", "k", "l", "m", "n", "o", "p", "q", "r",
  "s", "t", "u", "v", "w", "x", "y", "z",
];

/* ===============
TOPICS
=============== */
export const TOPICS = {
  movies,
  sport,
  words,
};

/* ===============
THEMES
=============== */
export const THEMES = {
  dark: darkTheme,
  light: lightTheme,
};
