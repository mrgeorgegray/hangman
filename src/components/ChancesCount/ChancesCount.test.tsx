import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "@emotion/react";

import ChancesCount from ".";
import { THEMES } from "../../config";

describe("<ChancesCount />", () => {
  it("renders", () => {
    const { container } = render(
      <ThemeProvider theme={THEMES.light}>
        <ChancesCount chancesRemaining={2} />
      </ThemeProvider>
    );

    expect(container).toMatchSnapshot();
  });
});
