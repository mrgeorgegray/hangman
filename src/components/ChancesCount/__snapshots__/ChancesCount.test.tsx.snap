// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChancesCount /> renders 1`] = `
.emotion-0 {
  font-size: 14px;
  margin-bottom: 16px;
}

.emotion-1 {
  color: #949DA5;
}

<div>
  <div
    class="emotion-0"
  >
    <p
      aria-live="polite"
      class="emotion-1"
      id="chancesRemaining"
    >
      (chances remaining: 
      2
      )
    </p>
  </div>
</div>
`;
